pkg/client/typed/discovery/discovery_client.go

```diff
@@ -67,6 +67,10 @@ type ServerResourcesInterface interface {
 	ServerResourcesForGroupVersion(groupVersion string) (*unversioned.APIResourceList, error)
 	// ServerResources returns the supported resources for all groups and versions.
 	ServerResources() (map[string]*unversioned.APIResourceList, error)
+	// ServerAllResources returns the supported resources for all groups and versions.
+	ServerAllResources() ([]unversioned.GroupVersionResource, error)
+	// ServerAllNamespacedResources returns the supported namespaced resources for all groups and versions.
+	ServerAllNamespacedResources() ([]unversioned.GroupVersionResource, error)
 	// ServerPreferredResources returns the supported resources with the version preferred by the
 	// server.
 	ServerPreferredResources() ([]unversioned.GroupVersionResource, error)
```

Comment:
I still don't understand why we aren't including the resources that only exist in non-preferred versions in the result of this function. anyone using this method today is missing resources. if a resource exists only in non-preferred versions, then I think we should pick something reasonable (like first version found), but still include the resource